Convert flag data to json file

flickz opened the pull request:
Ref #875

flickz commented:
@louh This PR is not ready yet. (Right now, I have created `flags.json` file in `app/data/`). I want to be sure if it's also requires creating a `GET` API, endpoint `/flags/:name` to fetch flags from the front end. Just like I listed the steps [here](https://github.com/streetmix/streetmix/issues/875#issuecomment-387114386) and how locale works.

flickz commented:
No, I'm currently writing the test and API code

flickz commented:
@louh I have the test and API now

louh commented on app/data/flags.json:
```diff
@@ -0,0 +1,48 @@
+  },
+  "SEGMENT_ANGLED_PARKING": {
+    "label": "Segment — angled parking",
+    "defaultValue": false,
```
Are you working from the latest version of the master branch? For instance, the current `https://github.com/streetmix/streetmix/blob/master/assets/scripts/app/flag_data.js` has the `defaultValue` for `SEGMENT_ANGLED_PARKING` set to `true`. Please go through and verify that this json file matches all the properties in `flag_data.js` exactly.

To remove confusion, let's also remove `flag_data.js` after it's been ported here, then update any scripts that use it: https://github.com/streetmix/streetmix/search?q=flag_data&unscoped_q=flag_data

For now, just import the file directly when compiling the front-end, don't use the API yet.

louh commented on app/resources/v1/flags.js:
```diff
@@ -0,0 +1,40 @@
+
+const readFile = util.promisify(fs.readFile)
+
+async function getFlags (res) {
```
Instead of `getFlags` let's rename to `readFlags` to better reflect what this function is doing.

flickz commented:
@louh I have updated the [`flags.json`](https://github.com/streetmix/streetmix/blob/2fcdf716d67ba21432301eaccacc0280c79da9c2/app/data/flags.json) to match latest [`flag_data.js`](https://github.com/streetmix/streetmix/blob/master/assets/scripts/app/flag_data.js) in master. 
Also updated all front-end import to directly use flags.json but some test are failing. Please help me take a look

louh commented:
Sure. Let me try to walk you through the process of fixing the tests.

1. First, look at the Travis CI build details. Scroll down to [where the tests are failing](https://travis-ci.org/streetmix/streetmix/builds/377839552#L4279) and verify that the test is failing for a legitimate reason. (Sometimes, tests fail because a service timed out, in which case we just restart the test.)
2. To further verify the tests are failing legitimately, run `npm test` locally. If Jest seems to pass immediately, it's probably because it didn't test all the files. You can "Press p to filter by a filename regex pattern." and test specifically for the failing test suite, which in this case is `flags.test.js` (just typing `flags` is probably good enough to run the suite)
3. Look at the expected value and the return value. Note that the return value is not what is expected.
4. It's most likely that the cause is that when the format was changed to `json`, the data structure was also changed somewhat. Although the mock path has updated, the [mock return value](https://github.com/streetmix/streetmix/blob/flag-data-json/assets/scripts/store/reducers/__tests__/flags.test.js#L5-L18) has not changed.
5. Try revising the mock's return value. Does the test pass now?

flickz commented:
Yeah it's passed!
